server/service/apple_mdm.go

```diff
@@ -4152,3 +4152,61 @@ func (svc *Service) DisableABM(ctx context.Context) error {
 	appCfg.MDM.AppleBMEnabledAndConfigured = false
 	return svc.ds.SaveAppConfig(ctx, appCfg)
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// GET /ota
+////////////////////////////////////////////////////////////////////////////////
+
+type getOTAProfileRequest struct {
+	EnrollSecret string `query:"enroll_secret"`
+}
+
+type getOTAProfileResponse struct {
+	Profile string `json:"profile"`
+	Err     error  `json:"error,omitempty"`
+}
+
+func (r getOTAProfileResponse) error() error { return r.Err }
+
+func getOTAProfileEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (errorer, error) {
+	req := request.(*getOTAProfileRequest)
+	profile, err := svc.GetOTAProfile(ctx, req.EnrollSecret)
+	if err != nil {
+		return &getOTAProfileResponse{Err: err}, err
+	}
+
+	reader := bytes.NewReader(profile)
+	return &getMDMAppleConfigProfileResponse{fileReader: io.NopCloser(reader), fileLength: reader.Size(), fileName: "foobar.mobileconfig"}, nil
+}
+
+func (svc *Service) GetOTAProfile(ctx context.Context, enrollSecret string) ([]byte, error) {
+	// Skip authz as this endpoint is used by end users from their iPhones or iPads; authz is done
+	// by the enroll secret verification below
+	svc.authz.SkipAuthorization(ctx)
+
+	_, err := svc.ds.VerifyEnrollSecret(ctx, enrollSecret)
+	if err != nil {
+		if fleet.IsNotFound(err) {
+			return nil, fleet.NewAuthFailedError("invalid enroll secret for OTA profile")
```

Comment:
Will this include a rate limiter to mitigate brute force attempts to get enroll secrets?

Reply:
That's a good idea; it wasn't spec'd, but I can look into it if we want it.

Reply:
sounds like a good idea, search in `handler.go` for inspiration we already have implemented rate limits for other endpoints.

Reply:
@roperzh or @gillespi314 any thoughts on the rate quota? I went with the same values as the rate limiter for logging in, since that seemed most analogous to what I'm doing here, but lmk if there's something better.

Reply:
sgtm!

Reply:
hm, wait. Thinking more about it, could somebody block all ipads from enrolling just by intentionally hitting the rate limit?

Reply:
@roperzh ah that's a good point... yeah that would definitely happen

Reply:
should we pass on the rate limiter for now then?

Reply:
> hm, wait. Thinking more about it, could somebody block all ipads from enrolling just by intentionally hitting the rate limit?

I thought the limit store was smarter than that? Doesn't it keep track of the requestor?

Reply:
> I thought the limit store was smarter than that? Doesn't it keep track of the requestor?

@gillespi314 I just tried (hit the limit on my workstation, and then sent a request from a different machine), and it seems like it doesn't. I could be missing something tho!

Reply:
Although looking at the code it does seem like it should be tracking the requestor: https://github.com/fleetdm/fleet/blob/main/server/service/middleware/ratelimit/ratelimit.go#L89

Reply:
Huh, yeah, looking at the code, it isn't limiting by IP address or anything, just limiter for each route.

Reply:
should I revert the limiting changes for now? Perhaps we can address it in a follow up (since we'll need to change the rate limiting code ig)?